Source/ASTextNode2.mm

```diff
@@ -360,115 +366,65 @@ - (NSObject *)drawParametersForAsyncLayer:(_ASDisplayLayer *)layer
   [self _ensureTruncationText];
   ASTextContainer *copiedContainer = [_textContainer copy];
   copiedContainer.size = self.bounds.size;
-  NSMutableAttributedString *mutableText = [self.attributedText mutableCopy] ?: [[NSMutableAttributedString alloc] init];
-  
-  [self prepareAttributedString:mutableText];
   
   return @{
-    @"container": copiedContainer,
-    @"text": mutableText,
-    @"bgColor": self.backgroundColor ?: [NSNull null]
-  };
+           @"container": copiedContainer,
+           @"text": [self l_preparedAttributedText],
+           @"bgColor": self.backgroundColor ?: [NSNull null],
+           // Pass along self as a hack so that we can update _lastUsedLayout.
+           // We'll be careful
+           @"instance" : self
```

Comment:
Hmm, lol. Should we use ASWeakProxy, or is that guaranteed to not make a difference in when we're deallocated?

Since it is possible to have two displays going on at the same time (the sentinel in ASDisplayNode tracks which one should win), I'm not sure it is safe to set _lastUsedLayout unless we're checking which display is intended to / applied as the winner?

Reply:
Yeah it's definitely possible for the loser to be set – but `lastUsedLayout` is not considered authoritative. As the comment says, it's just a fast-path for the common case – a first place to check before going to the global layout cache. If we miss, it's not a tragedy.